### ISSUE TITLE
Strange error with --rseed flag

### ISSUE THREAD

**Issue by mattstepney** [sirocco-rt/sirocco#920]
When running a model on multiple cores (120) and lots of photons (7e8), if you mistype the random seed flag (in my case "-rseed") you get an inelegant/confusing error. The simulation immediately halts, but instead of displaying a page similar to that of py -h, I received a "bus error (core dumped)" message. Nick believes he knows the cause of the error (Something to do with memory overload on the zeroth core)

**Comment by jhmatthews** [comment]
Thanks Matt. I don't know if I necessarily can reproduce the bus error, but I can confirm if I run ```py -rsee root.pf``` I get an MPI abort message saying things haven't been wrapped up properly, i.e. along the lines of

```
--------------------------------------------------------------------------
mpirun has exited due to process rank 3 with PID 0 on
node lamd031 exiting improperly. There are three reasons this could occur:
```

I could certainly imagine with loads of cores on a cluster you could get a similar error. This happens because we use exit() rather than Exit(), and we print out for all threads. I would suggest we adjust the help function to look like this:
```C
void
help ()
{
#ifdef MPI_ON
  if (rank_global == 0)
  {
#endif
    char *some_help;

    some_help = "\
\n\
This program simulates radiative transfer in a (biconical) CV, YSO, quasar, TDE or (spherical) stellar wind \n\
\n\
Usage:  py [-h] [-r] [-t time_max] [-v n] [--dry-run] [-i] [--version] [--rseed] [-p n_steps] [-classic] [-srclassic] xxx  or simply py \n\
\n\
where xxx is the rootname or full name of a parameter file, e. g. test.pf \n\
\n\
and the switches have the following meanings \n\
\n\
 -h             Print this help message and stop \n\
 -r             Restart a run of the progarm reading the file xxx.windsave \n\
 -t time_max    Limit the total time to approximately time_max seconds.  Note that the program checks \n\
                for this limit somewhat infrequently, usually at the ends of cycles, because it \n\
                is attempting to save the program outputs so that the program can be restarted with \n\
                -r if that is desired. \n\
 -v n           Increase or decrease the amount of print out.  The default is 5.  Larger numbers increase  \n\
                the amount printed; smaller numbers decrease it.   \n\
 --dry-run      Create a new .pf file and stop \n\
 -i             Same as --dry-run \n\
 --version      Print out python version, commit hash and if there were files with uncommitted \n\
                changes and stop \n\
 --rseed        Set the random number seed to be time-based, rather than fixed. \n\
 --rng          Save or load the RNG state to file, to allow persistent RNG states between restarts\n\
\n\
Other switches exist but these are not intended for the general user.\n\
These are largely diagnostic or for special cases. These include\n\
 -d             Enable advanced/diagnostic inputs (normally for debugging purposes) \n\
                Python will then query the user for information about what to do with a series of \n\
                inputs beginning with @ \n\
 -e             Change the maximum number of errors of one type (by default 100,000) before the program will quit\n\
 -e_write 	Change the maximum number of errors of one type (by default 100) to print out before recording errors silently\n\
 -f             Invoke a fixed temperature mode, used for runs with Zeus or Plutu \n\
 -z             Invoke a special mode for that causes Python to start with a run from Zeus or Plutu\n\
 -p [range]     Vary the number of photons in ionization cycles logarthmically building up to the final value\n\
                Range is in powers of 10, the difference beween the number of photons in the first cycle \n\
                compared to the last. If range is missing, range is assumed to be 1, in which case the  \n\
                number of photons will in the first cycle will be one order of magniude less than in the last cycle \n\
 -classic       Use Python in it's classic configuration, with linear Doppler shifts, etc., and where co-moving frame\n\
                effects are not taken into account.\n\
 -srclassic     Use Python with full special relativity for Doppler shits, etc., but do not include any co-moving frame\n\
                effects.\n\
\n\
If one simply types py or pyZZ where ZZ is the version number, one is queried for a name \n\
of the parameter file and inputs will be requested from the command line. \n\
\n\
\n\
";                              // End of string to provide one with help

    printf ("%s\n", some_help);
#ifdef MPI_ON
  }
#endif

  Exit (0);                     
}
```

i.e. so it only prints out the help function for thread 0, and the Exit() function ensures that MPI is wrapped up properly with MPI_Abort. There is a comment in there saying we want to use exit() rather than Exit(), but I don't know what the motivation for that is. Any problem with the above fix @Higginbottom or @kslong ?

**Comment by Higginbottom** [comment]
I certainly don't have a problem with this fix!!

**Comment by kslong** [comment]
This looks reasonable to me.

**Comment by jhmatthews** [comment]
Should be fixed by #922. Reopen if not!
